Improve some CSS WPT tests so their output is easier to read and fix some mistaken expectations

darinadler opened the pull request:
#### e18f26fb8c87bba56e15e584dbc6a0fe7538cd48
<pre>
Improve some CSS WPT tests so their output is easier to read and fix some mistaken expectations
<a href="https://bugs.webkit.org/show_bug.cgi?id=248817">https://bugs.webkit.org/show_bug.cgi?id=248817</a>
rdar://problem/103026393

Reviewed by Tim Nguyen.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/cssom-getBoxQuads-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/cssom-getBoxQuads-001.html:
Update to be a set of separate tests instead of a single test with a lot of assertions.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/flex-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/flex-serialization.html:
Ditto.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/overflow-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/overflow-serialization.html:
Ditto.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values.html:
Fixed the test to cover a case that should work where the longhand has the initial value rather
than a case that should not where the longhand has the initial keyword as its value. WebKit is now
failing this test case.

Canonical link: <a href="https://commits.webkit.org/257438@main">https://commits.webkit.org/257438@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/717d68457ca04cab7f5b219f97bf9fb3800d67fb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/99009 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8215 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32162 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108415 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/168667 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8768 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/85553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91532 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/106373 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104730 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/6634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/90226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/33662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/88456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21571 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/76516 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/2116 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/23087 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/2020 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/85553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5120 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6983 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/42562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3434 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/cssom-getBoxQuads-001.html:
```diff
@@ -33,19 +33,29 @@
-      let fb = document.getElementById("flex-block");
-      assert_equals(fb.getBoxQuads({box: "border"})[0].getBounds().width,  20, "Flex layout border box is expected width.");
-      assert_equals(fb.getBoxQuads({box: "margin"})[0].getBounds().width, 100, "Flex layout margin box is expected width.");
+    // For containers that expand items to fill block-axis space, measure the box heights also.
+    test(function() {
+      assert_equals(fb.getBoxQuads({box: "border"})[0].getBounds().width, 20)
+    }, "Flex layout border box is expected width.")
+    test(function() {
+      assert_equals(fb.getBoxQuads({box: "margin"})[0].getBounds().width, 100)
+    }, "Flex layout margin box is expected width.")
 
-      assert_equals(fb.getBoxQuads({box: "border"})[0].getBounds().height, 10, "Flex layout border box is expected height.");
-      assert_equals(fb.getBoxQuads({box: "margin"})[0].getBounds().height, 50, "Flex layout margin box is expected height.");
```
Fwiw, you can also use the `single_test` setup, which would let you avoid calling test() everytime: https://web-platform-tests.org/writing-tests/testharness-api.html#single-page-tests

Same for the other test.

```
<title>Flex layout margin box is expected height</title>
<script>
    setup({ single_test: true });
    // For containers that expand items to fill block-axis space, measure the box heights also.
    let fb = document.getElementById("flex-block");
    assert_equals(fb.getBoxQuads({box: "border"})[0].getBounds().width,  20, "Flex layout border box is expected width.");
    assert_equals(fb.getBoxQuads({box: "margin"})[0].getBounds().width, 100, "Flex layout margin box is expected width.");

    assert_equals(fb.getBoxQuads({box: "border"})[0].getBounds().height, 10, "Flex layout border box is expected height.");
    assert_equals(fb.getBoxQuads({box: "margin"})[0].getBounds().height, 50, "Flex layout margin box is expected height.");
    done();
</script>
```

darinadler replied:
Sure, that seems better for adapting these tests.

nt1m replied:
Actually I just realized `single_test` will probably not use the output you want :( 

I'm fine with the current patch as it is too fwiw.

nt1m reviewed (approved):
Looks fine, but you might want to consider using the `setup({ single_test: true });`, which is a lot shorter than repeating `test()` everytime.

darinadler committed:
Improve some CSS WPT tests so their output is easier to read and fix some mistaken expectations
https://bugs.webkit.org/show_bug.cgi?id=248817
rdar://problem/103026393

Reviewed by Tim Nguyen.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/cssom-getBoxQuads-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/cssom-getBoxQuads-001.html:
Update to be a set of separate tests instead of a single test with a lot of assertions.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/flex-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/flex-serialization.html:
Ditto.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/overflow-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/overflow-serialization.html:
Ditto.

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom/shorthand-values.html:
Fixed the test to cover a case that should work where the longhand has the initial value rather
than a case that should not where the longhand has the initial keyword as its value. WebKit is now
failing this test case.

Canonical link: https://commits.webkit.org/257438@main
